### PR TITLE
Change Raw encoding to use Bytebuffer

### PR DESCRIPTION
**Pull request by junchaowu**
-Change the encodingType to encoding
-Add generic constrains into the RPC client and serializer
-Change send method name to call
-Add this to all instance method or variable references.
-Fix minor issues

### CODE REVIEW

File: yarpc-core/src/main/java/io/yarpc/RPCFuture.java [hunk 1]
```diff
@@ -48,6 +48,6 @@ public boolean setException(Throwable throwable) {
     }
 
     public void sync() throws ExecutionException, InterruptedException {
-        get();
+        this.get();
```
Comment (anson627) [inline yarpc-core/src/main/java/io/yarpc/RPCFuture.java]: not sure what this function do? a wrapper on top of one line code?

**Comment by anson627** [comment]
lgtm, one general recommendation, try to avoid generic, unless have to. It makes both library users and authors' lives.

**Comment by breerly** [comment]
Tests are failing - also, looks like you'll want these to be separate PR's for each item here

**Comment by junchaowu** [comment]
abandon this. Will submit into different small PR's

**Pull request closed without merging**